core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -857,9 +862,11 @@ private[spark] class BlockManager(
 
     val startTimeMs = System.currentTimeMillis
     var blockWasSuccessfullyStored: Boolean = false
+    var exceptionWasThrown: Boolean = true
     val result: Option[T] = try {
       val res = putBody(putBlockInfo)
       blockWasSuccessfullyStored = res.isEmpty
+      exceptionWasThrown = false
       res
     } finally {
       if (blockWasSuccessfullyStored) {
```

Comment:
Wouldn't blockWasSuccessfullyStored be false if exceptionWasThrown were true ?
In that case, couldn't we write this as
try {
...
} catch (Exception e) {
  removeBlock
  addUpdatedBlock
} finally {
 // WHatever was there before ?
}